src/mongocrypt.c

```diff
@@ -124,6 +124,17 @@ mongocrypt_new (void)
    _mongocrypt_log_init (&crypt->log);
    crypt->ctx_counter = 1;
    crypt->cache_oauth_azure = _mongocrypt_cache_oauth_new ();
+   crypt->cache_oauth_gcp = _mongocrypt_cache_oauth_new ();
+
+   if (0 != _mongocrypt_once (_mongocrypt_do_init) ||
+       !(_native_crypto_initialized)) {
+      mongocrypt_status_t *status = crypt->status;
+
+      CLIENT_ERR ("failed to initialize");
+      /* Return crypt with failure status so caller can obtain error when
+       * calling mongocrypt_init */
+   }
+
```

Comment:
base64 decoding requires `_mongocrypt_do_init` to be called first. And now we base64 decode the GCP private key in `mongocrypt_setopt_kms_providers` before `mongocrypt_init`. So this now does the necessary global initialization when allocating a `mongocrypt_t`.